In [2]:
import pandas as pd

In [7]:
df = pd.read_csv("dataset_size_statistical_summary.csv")
true_rows = df[df["non_inferior"] == True]
true_rows.head(1)

,dataset_size,run_1_reward,run_2_reward,run_3_reward,run_4_reward,run_5_reward,average_reward,mean_reward,std_reward,sem_reward,ci_lower,ci_upper,non_inferior
21,440,494.12,492.56,473.28,499.56,498.0,491.504,491.504,10.573196,4.728477,478.375643,504.632357,True


To determine the optimal dataset size, I used **non-inferiority analysis**. The goal was to find the smallest dataset size that achieved performance statistically close to the best observed performance. 

For each dataset size, the imitation learning model was trained five independent times. I calculated the mean reward, standard deviation, and 95% confidence intervals. I then checked whether each dataset size was non-inferior to the best observed mean reward.

The non-inferiority margin was set to 25 reward points. Since the maximum reward in CartPole-v1 is 500, this margin corresponds to 5% of the maximum possible reward. A dataset size was considered non-inferior if the lower bound of its 95% confidence interval was within 25 reward points of the best observed mean reward (which is 475 rewards).

The first dataset size that satisfied the non-inferiority criterion was 440 expert demonstrations. At this dataset size, the five reward values were:494.12, 492.56, 473.28, 499.56, 498.0
The average reward was 491.504, with a standard deviation of 10.573. The 95% confidence interval was approximately 478.376 to 504.632. Since the lower bound of the confidence interval is above the non-inferiority threshold, 440 demonstrations were considered sufficient.

Therefore, the optimal dataset size was selected as 440 expert demonstrations. This dataset size achieves near-maximum CartPole performance while using fewer demonstrations than larger datasets.